### ISSUE TITLE
32-bit Firefox fails to launch with --chaos

### ISSUE THREAD

**Issue by tysmith** [rr-debugger/rr#3423]
When trying to launch a 32-bit build of Firefox I see the following

```
ERROR: ld.so: object '/usr/local/bin/../lib/rr/librrpreload_32.so' from LD_PRELOAD cannot be preloaded (failed to map segment from shared object): ignored.
ERROR: ld.so: object 'libdl.so.2' from LD_PRELOAD cannot be preloaded (failed to map segment from shared object): ignored.
/home/user/workspace/browsers/linux32-m-c-20230112213033-fuzzing-asan-opt/firefox: error while loading shared libraries: libpthread.so.0: failed to map segment from shared object
```

The command used
```
rr record --chaos --disable-cpuid-features-ext 0xdc230000,0x2c42,0xc /home/user/workspace/browsers/linux32-m-c-20230112213033-fuzzing-asan-opt/firefox -no-remote
```

If you'd like to try to reproduce locally builds are available here: https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/gecko.v2.mozilla-central.latest.firefox.linux-fuzzing-asan-opt/artifacts/public%2Fbuild%2Ftarget.tar.bz2

**Comment by tysmith** [comment]
I guess I should confirm that if I remove `--chaos` it launches as expected.

**Comment by khuey** [comment]
Which rr rev are you using?

**Comment by tysmith** [comment]
The latest available code in git.

**Comment by khuey** [comment]
Do you see this behavior on fbdceb04b31025c4120f31f451450c080570abb3?

**Comment by tysmith** [comment]
Yes I can still repro with fbdceb0.

**Comment by bernhardu** [comment]
It looks like the above firefox 32-bit ASAN build did run before commit 143ebdd4fc7.
Reverting that commits lets the process continue over the librrpreload_32.so loading,
but fails later after an rrcall_init_buffers syscall.

A part of that patch raises the question if this is still expected to work?
https://github.com/rr-debugger/rr/blob/c88aaa7f4bf9611925cc7d9f4bda157b0e956ab5/src/AddressSpace.cc#L2298-L2299

**Comment by rocallahan** [comment]
Getting that to work is hard and I don't think it's that important. What we can do is, for 32-bit ASAN builds, don't bother trying to randomize memory allocations in chaos mode.

**Comment by rocallahan** [comment]
Fixed by 714db6a35a5a18ef793af73b9c70fca42a2dc2e7
